### PR TITLE
Update latest chrono

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

New chrono, new deprecations.

This just does the bare minimum to get the CI green, follow on work is tracked in #4456 and #5480

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-array/src/temporal_conversions.rs [hunk 1]
```diff
@@ -130,45 +131,48 @@ pub fn time_to_time64ns(v: NaiveTime) -> i64 {
 /// converts a `i64` representing a `timestamp(s)` to [`NaiveDateTime`]
 #[inline]
 pub fn timestamp_s_to_datetime(v: i64) -> Option<NaiveDateTime> {
-    NaiveDateTime::from_timestamp_opt(v, 0)
+    Some(DateTime::from_timestamp(v, 0)?.naive_utc())
```
Comment (tustvold) [inline arrow-array/src/temporal_conversions.rs:134]: https://github.com/apache/arrow-rs/issues/5480 tracks cleaning up these APIs to return `DateTime<Utc>`

**Comment by tustvold** [comment]
Merging this as the changes are mechanical and this is blocking the CI pipelines

File: arrow-array/src/temporal_conversions.rs [hunk 2]
```diff
@@ -179,13 +183,13 @@ pub(crate) fn split_second(v: i64, base: i64) -> (i64, u32) {
 /// converts a `i64` representing a `duration(s)` to [`Duration`]
 #[inline]
 pub fn duration_s_to_duration(v: i64) -> Duration {
-    Duration::seconds(v)
+    Duration::try_seconds(v).unwrap()
```
Comment (alamb) [inline arrow-array/src/temporal_conversions.rs:186]: is there any performance implication here (e.g. starting to check for invalid durations when we didn't before)?

**Review by alamb — COMMENTED** [review]
Thank you @tustvold  -- I took a look at this PR and the changes make sense to me

File: arrow-array/src/temporal_conversions.rs (hunk 2 above)
Comment (tustvold) [reply arrow-array/src/temporal_conversions.rs:186]: No, it just moves the panic

**Pull request merged**